pkg/grpc/exec/exec.go

```diff
@@ -199,14 +211,36 @@ func (msg *MsgExecveEventUnix) Retry(internal *process.ProcessInternal, ev notif
 
 	// Check we have a parent with exception for pid 1, note we do this last because we want
 	// to ensure the podInfo and process are set before returning any errors.
-	if proc.Pid.Value > 1 && parent == nil {
-		parentId := proc.ParentExecId
+	if tetragonProcess.Pid.Value > 1 && tetragonParent == nil {
+		parentId := tetragonProcess.ParentExecId
 		parent, err := process.Get(parentId)
-		if parent == nil {
+		if parent != nil {
+			tetragonParent = parent.UnsafeGetProcess()
+			parent.RefInc("parent")
+			ev.SetParent(tetragonParent)
+		} else {
+			eventcache.CacheRetries(eventcache.ParentInfo).Inc()
 			return err
 		}
-		parent.RefInc("parent")
-		ev.SetParent(parent.UnsafeGetProcess())
```

Comment:
same here, it looks like we can keep the same path as original code, just add eventcache.CacheRetries to the error path?